src/vector/ops/geometry/intersection.rs

```diff
@@ -0,0 +1,74 @@
+use crate::vector::Geometry;
+use gdal_sys::OGR_G_Intersection;
+
+/// An intersection between Geometry/Geometry returning the same type.
+pub trait Intersection
+where
+    Self: Sized,
+{
+    /// Compute intersection.
+    ///
+    /// Generates a new geometry which is the region of intersection of
+    /// the two geometries operated on. Call intersects (Not yet implemented)
+    /// to check if there is a region of intersection.
+    /// Geometry validity is not checked. In case you are unsure of the
+    /// validity of the input geometries, call IsValid() before,
+    /// otherwise the result might be wrong.
+    ///
+    /// # Returns
+    /// Ok(Geometry) if the two geometries intersect
```

Comment:
method returns an `Option<Geometry>`, but this documentation shows as if it returns a `Result<Geometry>`

Reply:
Updated. And added another unit test.